OpenRA.Mods.Common/Traits/BotModules/McvManagerBotModule.cs

```diff
@@ -121,16 +121,20 @@ void IBotTick.BotTick(IBot bot)
 			}
 		}
 
-		bool ShouldBuildMCV()
+		int CountConstructionYards(Player owner)
+		{
+			return AIUtils.CountBuildingByCommonName(Info.ConstructionYardTypes, owner);
+		}
+
+		bool AllowedToBuildMCV()
```

Comment:
I renamed it to `AllowedToBuildMCV` because it handles also the check for `MaxNumberOfMCV`